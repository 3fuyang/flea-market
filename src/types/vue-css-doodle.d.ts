declare module '@luxdamore/vue-css-doodle'