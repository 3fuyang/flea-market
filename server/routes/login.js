// Login页面的接口
const express = require('express')
const app = express()

const connection = require('../database/db')

app.use(express.json())
app.use(express.urlencoded({extended:  false}))

// 接口1 用户登录：传入（账号、密码） 返回（登录结果）
app.post('/userlogin', (req, res) => {
  let doesExist = false
  connection.query(
    `select count(*) as cnt from userAccount where user_id = ? and mypassword = ?` ,
    [req.body.id, req.body.password],
    (err, result) => {
      if (err) throw err
      doesExist = (result[0].cnt > 0) ? true : false
      res.end(JSON.stringify(doesExist))
    }
  )
})

// 接口2 管理员登录：传入（账号、密码） 返回（登录结果）
app.post('/adminlogin', (req, res) => {
  let doesExist = false
  connection.query(
    `select count(*) as cnt from adminAccount where user_id = ? and mypassword = ?`, 
    [req.body.id, req.body.password],
    (err, result) => {
      if (err) throw err
      doesExist = (result[0].cnt > 0) ? true : false
      res.end(JSON.stringify(doesExist))
    }
  )
})

// 接口3 校验学号或手机号是否已被注册：传入（学号、手机号） 返回（是否已被注册）
app.post('/isrepeated', (req, res) => {
  let isRepeated = false
  connection.query(
    `select count(*) as cnt from userAccount where user_id = ? or telnum = ?`, 
    [req.body.id, req.body.telnum],
    (err, result) => {
      if (err) throw err
      isRepeated = (result[0].cnt > 0) ? true : false
      res.end(JSON.stringify(isRepeated))
    }
  )
})

// 接口4 用户注册：传入（账号、密码、姓名、手机号、性别、学院、生日） 返回（null）
app.post('/register', (req, res) => {
  let newAccount = [
    req.body.user_id,
    req.body.mypassword,
    req.body.myname,
    req.body.telnum,
    req.body.gender,
    req.body.college,
    req.body.birthday,
    req.body.myname
  ]
  connection.query(
    "insert into userAccount(user_id,mypassword,real_name,telnum,gender,college,birthday,nickname) value(?,?,?,?,?,?,?,?)", 
    newAccount, 
    (err, result) => {
      if (err) throw err
      res.end(JSON.stringify(result))
    }
  )
})

// 接口5 校验学号和手机号是否已被注册且匹配：传入（学号、手机号） 返回（是否已被注册且匹配）
app.post('/idcoupletel', (req, res) => {
  let isCoupled = false
  connection.query(
    `select count(*) as cnt from userAccount where user_id = ? and telnum = ?`,
    [req.body.id, req.body.telnum],
    (err, result) => {
      if (err) throw err
      isCoupled = (result[0].cnt > 0) ? true : false
      res.end(JSON.stringify(isCoupled))
    }
  )
})

// 接口6 用户修改密码：传入（ID、新密码） 返回（null）
app.post('/modifypassword', (req, res) => {
  connection.query(
    `update userAccount set mypassword = ? where user_id = ?`,
    [req.body.newpassword, req.body.id],
    (err, result) => {
      if (err) throw err
      res.end(JSON.stringify(result))
    }
  )
})

module.exports = app 