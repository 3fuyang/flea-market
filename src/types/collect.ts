export class Collect {
  userID: string = ''
  goodID: string = ''
  collected: boolean = false
  collect () {
    if (!this.userID) {
      return '请先登录'
    } else if (this.userID.length === 4) {
      return '管理员请登录普通账号进行收藏'
    } else if (!this.goodID) {
      return '商品不存在'
    } else if (this.collected) {
      return '不能重复收藏'
    } else {
      return true
    }
  }
  set (uid: string = '', gid: string = '', collected: boolean = true) {
    this.userID = uid
    this.goodID = gid
    this.collected = collected
  }
}