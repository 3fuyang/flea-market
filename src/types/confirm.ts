export class Confirm {
  orderID: string = '1'
  stat: '待付款' | '待确认' | '待评价' | '已完成' = '待确认'
  confirm () {
    switch (this.stat) {
      case '待付款':
        return '请先付款'
      case '待确认':
        return true
      default:
        return '不可重复确认'
    }
  }
  set (st: '待付款' | '待确认' | '待评价' | '已完成' = '待确认') {
    this.stat = st
  }
}