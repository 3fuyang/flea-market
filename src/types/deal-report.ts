export class DealReport {
  adminID: string = '0001'
  orderID: string = '1'
  reported: '待处理' | '已封禁' | '已驳回' = '待处理'
  reply: string = ''
  dealReport (stat: '待处理' | '已封禁' | '已驳回') {
    if (this.reported !== '待处理') {
      return '不可重复处理举报'
    } else if (!this.reply) {
      return '请输入回复'
    } else {
      this.reported = stat
      return true
    }
  }
  set (rptd: '待处理' | '已封禁' | '已驳回', rpl: string) {
    this.reported = rptd
    this.reply = rpl
  }
}