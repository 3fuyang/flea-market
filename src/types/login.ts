// 测试用，类型建模
export class LoginForm {
  // 类型
  type: 'member' | 'admin' = 'member'
  // 用户名
  userID: string = ''
  // 密码
  password: string = ''
  // 登录
  login () {
    if (!this.userID) {
      return '请输入账号'
    } else if (this.type === 'member' && this.userID.length !== 7) {
      return '普通用户账号应长7位'
    } else if (this.type === 'admin' && this.userID.length !== 4) {
      return '管理员账号应长4位'
    } else if (!this.password) {
      return '请输入密码'
    } else {
      // 登录
      return true
    }
  }
  // setter
  set (tp: 'member' | 'admin' = 'member', id: string = '', pwd: string = '') {
    this.type = tp
    this.userID = id
    this.password = pwd
  }
}