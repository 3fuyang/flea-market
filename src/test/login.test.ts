import { describe, expect, it } from 'vitest'
import { LoginForm } from '@/types/login'

const loginForm = new LoginForm()

// describe creates a suite
describe('login', () => {
  // it is an alias of test
  it('li_1', () => {
    loginForm.set('member', '123', '')
    expect(loginForm.login()).toBe('普通用户账号应长7位')
    expect(loginForm).toMatchSnapshot()
  })

  it('li_2', () => {
    loginForm.set('member', '1234', '')
    expect(loginForm.login()).toBe('普通用户账号应长7位')
    expect(loginForm).toMatchSnapshot()    
  })

  it('li_3', () => {
    loginForm.set('member', '12345', '')
    expect(loginForm.login()).toBe('普通用户账号应长7位')
    expect(loginForm).toMatchSnapshot()
  })  

  it('li_4', () => {
    loginForm.set('member', '1234567', '')
    expect(loginForm.login()).toBe('请输入密码')
    expect(loginForm).toMatchSnapshot()
  })
  
  it('li_5', () => {
    loginForm.set('member', '12345678', '')
    expect(loginForm.login()).toBe('普通用户账号应长7位')
    expect(loginForm).toMatchSnapshot()
  }) 
  
  it('li_6', () => {
    loginForm.set('' as 'admin', '', '')
    expect(loginForm.login()).toBe('请输入账号')
    expect(loginForm).toMatchSnapshot()
  })  

  it('li_7', () => {
    loginForm.set('member', '123', 'qwe')
    expect(loginForm.login()).toBe('普通用户账号应长7位')
    expect(loginForm).toMatchSnapshot()
  })

  it('li_8', () => {
    loginForm.set('member', '1234', 'qwe')
    expect(loginForm.login()).toBe('普通用户账号应长7位')
    expect(loginForm).toMatchSnapshot()
  })

  it('li_9', () => {
    loginForm.set('member', '12345', 'qwe')
    expect(loginForm.login()).toBe('普通用户账号应长7位')
    expect(loginForm).toMatchSnapshot()
  })

  it('li_10', () => {
    loginForm.set('member', '1234567', 'qwe')
    expect(loginForm.login()).toBeTruthy()
    expect(loginForm).toMatchSnapshot()
  })

  it('li_11', () => {
    loginForm.set('member', '12345678', 'qwe')
    expect(loginForm.login()).toBe('普通用户账号应长7位')
    expect(loginForm).toMatchSnapshot()
  })

  it('li_12', () => {
    loginForm.set('admin', '123', '')
    expect(loginForm.login()).toBe('管理员账号应长4位')
    expect(loginForm).toMatchSnapshot()
  })

  it('li_13', () => {
    loginForm.set('admin', '1234', '')
    expect(loginForm.login()).toBe('请输入密码')
    expect(loginForm).toMatchSnapshot()
  })

  it('li_14', () => {
    loginForm.set('admin', '12345', '')
    expect(loginForm.login()).toBe('管理员账号应长4位')
    expect(loginForm).toMatchSnapshot()
  })  

  it('li_15', () => {
    loginForm.set('admin', '1234567', '')
    expect(loginForm.login()).toBe('管理员账号应长4位')
    expect(loginForm).toMatchSnapshot()
  })
  
  it('li_16', () => {
    loginForm.set('admin', '12345678', '')
    expect(loginForm.login()).toBe('管理员账号应长4位')
    expect(loginForm).toMatchSnapshot()
  }) 
  
  it('li_17', () => {
    loginForm.set('' as 'admin', '', '')
    expect(loginForm.login()).toBe('请输入账号')
    expect(loginForm).toMatchSnapshot()
  })  

  it('li_18', () => {
    loginForm.set('admin', '123', 'qwe')
    expect(loginForm.login()).toBe('管理员账号应长4位')
    expect(loginForm).toMatchSnapshot()
  })

  it('li_19', () => {
    loginForm.set('admin', '1234', 'qwe')
    expect(loginForm.login()).toBeTruthy()
    expect(loginForm).toMatchSnapshot()
  })

  it('li_20', () => {
    loginForm.set('admin', '12345', 'qwe')
    expect(loginForm.login()).toBe('管理员账号应长4位')
    expect(loginForm).toMatchSnapshot()
  })

  it('li_21', () => {
    loginForm.set('admin', '1234567', 'qwe')
    expect(loginForm.login()).toBe('管理员账号应长4位')
    expect(loginForm).toMatchSnapshot()
  })

  it('li_22', () => {
    loginForm.set('admin', '12345678', 'qwe')
    expect(loginForm.login()).toBe('管理员账号应长4位')
    expect(loginForm).toMatchSnapshot()
  })
})