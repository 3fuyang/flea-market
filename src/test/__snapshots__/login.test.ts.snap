// Vitest Snapshot v1

exports[`login > li_1 1`] = `
LoginForm {
  "password": "",
  "type": "member",
  "userID": "123",
}
`;

exports[`login > li_2 1`] = `
LoginForm {
  "password": "",
  "type": "member",
  "userID": "1234",
}
`;

exports[`login > li_3 1`] = `
LoginForm {
  "password": "",
  "type": "member",
  "userID": "12345",
}
`;

exports[`login > li_4 1`] = `
LoginForm {
  "password": "",
  "type": "member",
  "userID": "1234567",
}
`;

exports[`login > li_5 1`] = `
LoginForm {
  "password": "",
  "type": "member",
  "userID": "12345678",
}
`;

exports[`login > li_6 1`] = `
LoginForm {
  "password": "",
  "type": "",
  "userID": "",
}
`;

exports[`login > li_7 1`] = `
LoginForm {
  "password": "qwe",
  "type": "member",
  "userID": "123",
}
`;

exports[`login > li_8 1`] = `
LoginForm {
  "password": "qwe",
  "type": "member",
  "userID": "1234",
}
`;

exports[`login > li_9 1`] = `
LoginForm {
  "password": "qwe",
  "type": "member",
  "userID": "12345",
}
`;

exports[`login > li_10 1`] = `
LoginForm {
  "password": "qwe",
  "type": "member",
  "userID": "1234567",
}
`;

exports[`login > li_11 1`] = `
LoginForm {
  "password": "qwe",
  "type": "member",
  "userID": "12345678",
}
`;

exports[`login > li_12 1`] = `
LoginForm {
  "password": "",
  "type": "admin",
  "userID": "123",
}
`;

exports[`login > li_13 1`] = `
LoginForm {
  "password": "",
  "type": "admin",
  "userID": "1234",
}
`;

exports[`login > li_14 1`] = `
LoginForm {
  "password": "",
  "type": "admin",
  "userID": "12345",
}
`;

exports[`login > li_15 1`] = `
LoginForm {
  "password": "",
  "type": "admin",
  "userID": "1234567",
}
`;

exports[`login > li_16 1`] = `
LoginForm {
  "password": "",
  "type": "admin",
  "userID": "12345678",
}
`;

exports[`login > li_17 1`] = `
LoginForm {
  "password": "",
  "type": "",
  "userID": "",
}
`;

exports[`login > li_18 1`] = `
LoginForm {
  "password": "qwe",
  "type": "admin",
  "userID": "123",
}
`;

exports[`login > li_19 1`] = `
LoginForm {
  "password": "qwe",
  "type": "admin",
  "userID": "1234",
}
`;

exports[`login > li_20 1`] = `
LoginForm {
  "password": "qwe",
  "type": "admin",
  "userID": "12345",
}
`;

exports[`login > li_21 1`] = `
LoginForm {
  "password": "qwe",
  "type": "admin",
  "userID": "1234567",
}
`;

exports[`login > li_22 1`] = `
LoginForm {
  "password": "qwe",
  "type": "admin",
  "userID": "12345678",
}
`;
